bin/ch-pack-image

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+libexec=$(cd "$(dirname "$0")" && pwd)/../libexec/charliecloud
+. "${libexec}/base.sh"
+
+# shellcheck disable=SC2034
+usage=$(cat <<EOF
+Flatten a builder image into a Charliecloud packed file.
+
+Usage:
+
+  $ $(basename "$0") [-b BUILDER] [--fmt CH_PACK_FMT] IMAGE OUTDIR [ARGS ...]
+
+You must have sufficient privilege (via sudo) to run the Docker commands.
+EOF
+)
+
+parse_basic_args "$@"
+
+# Kludgey argument parsing.
+while true; do
```

Comment:
Can we not use the number of arguments, e.g., `$#`, in POSIX shell?

Reply:
ShellCheck is really good about flagging such things. If you're unsure, use the better syntax, and let ShellCheck gripe if you're wrong. Often that's easier than trying to look it up.